File: framework-cs/apps/UnityExample/Assets/Samples/Devian Foundation/0.1.0/GamePackage/Runtime/Ability/AbilityItemMaterial.cs
using Devian.Domain.Game;

namespace Devian
{
    public sealed class AbilityItemMaterial : AbilityItemBase
    {
        ITEM_MATERIAL mTable = null;

        public override string ItemId => mTable?.item_id ?? string.Empty;

        public void Init(ITEM_MATERIAL table)
        {
            mTable = table;
        }

        public override AbilityBase Clone()
        {
            var c = new AbilityItemMaterial();
            c.mTable = mTable;
            c.CopyStatsFrom(this);
            return c;
        }
    }
}

File: framework-ts/module/devian-domain-game/features/ability/AbilityItemMaterial.ts
import { ITEM_MATERIAL } from '../../Generated/Game.g';
import { AbilityItemBase } from './AbilityItemBase';

export class AbilityItemMaterial extends AbilityItemBase {
    private mTable: ITEM_MATERIAL | null = null;

    get itemId(): string { return this.mTable?.item_id ?? ''; }

    init(table: ITEM_MATERIAL): void {
        this.mTable = table;
    }

    clone(): AbilityItemMaterial {
        const c = new AbilityItemMaterial();
        c.mTable = this.mTable;
        c.copyStatsFrom(this);
        return c;
    }
}
